public/app/features/canvas/runtime/scene.tsx

```diff
@@ -133,18 +147,69 @@ export class Scene {
   };
 
   private findElementByTarget = (target: HTMLElement | SVGElement): ElementState | undefined => {
-    return this.root.elements.find((element) => element.div === target);
+    // We will probably want to add memoization to this as we are calling on drag / resize
```

Comment:
or revisit sticking the element directly on the dom node?

Reply:
That approach could work, but storing data directly in dom nodes typically isn't the best practice / messes up separation of concerns is bit of an anti pattern IMO

Reply:
generally agree with you -- aside from the fact that this function is only called from moveable/selecto and requires that we configured them with the same ref anyway